atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -0,0 +1,212 @@
+"""
+Copyright (c) 2016 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, unicode_literals
+
+try:
+    # py2
+    from ConfigParser import ConfigParser
+except ImportError:
+    # py3
+    from configparser import ConfigParser
+
+import json
+import re
+import os
+
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.koji_util import create_koji_session, TaskWatcher
+from dockerfile_parse import DockerfileParser
+
+
+class AddFilesystemPlugin(PreBuildPlugin):
+    """
+    Creates a base image by adding a filesystem generated by Koji
+
+    Submits an image build task to Koji based on kickstart.conf file
+    to create the file system to be used in creating the base image:
+    https://fedoraproject.org/wiki/Koji/BuildingImages#Building_Disk_Images
+
+    Once image build task is complete the tarball is downloaded and
+    it's imported into docker. This creates a new image. The existing
+    FROM instruction value is replaced with the ID of this new image.
+
+    The "FROM" instruction should be in the following format:
+        FROM scratch:<type>
+    Where type is one of the build types supported by this plugin,
+    currently only "kickstart" is supported.
```

Comment:
I don't remember which of us suggested it, it might have been me, but I wonder whether 'kickstart' is descriptive enough. How about this?:
```
FROM scratch:koji-image-build
```

Reasoning: the configuration file is not actually for kickstart directly (that's the .ks file it references), but for Koji's image-build task. Since that's the specific thing we're interacting with to get a base layer made I think we ought to name that rather than kickstart, which would most likely be common to any method of doing this.

What do you think?